### DISCUSSION TITLE
Hardware abilities database

### DISCUSSION THREAD

**Discussion by krahabb** [krahabb/meross_lan#13] [General]
Hello,
I would like to ask for help building a database of Meross devices features/capabilities in order to better understand how different devices could be categorized and characterized in order to help me integrate them into this component.
This would greatly drive my development without needing to have access to the physical devices in order to query it. Of course, issues and device behaviour will be hard to guess without experimenting message exchnage with the device but, at least, I would be able to 'mock' a reasonable implementation for the real device and, hopefully, with later minor fixes, correctly integrate it into this component.

What I ask for then, is to publish under this discussion the responses to these commands:

>POST http://ipof.your.meross.device/config HTTP/1.1
>Content-Type: application/json
>
>{
>    "header":{
>        "from":"",
>        "messageId":"",
>        "method":"GET",
>        "namespace":"Appliance.System.All",
>        "payloadVersion":1,
>        "sign":"cfcd208495d565ef66e7dff9f98764da",
>        "timestamp":0
>        },
>    "payload":{}
>}

>POST http://ipof.your.meross.device/config HTTP/1.1
>Content-Type: application/json
>
>{
>    "header":{
>        "from":"",
>        "messageId":"",
>        "method":"GET",
>        "namespace":"Appliance.System.Ability",
>        "payloadVersion":1,
>        "sign":"cfcd208495d565ef66e7dff9f98764da",
>        "timestamp":0
>        },
>    "payload":{}
>}

these can be sent with 'curl' or 'Rest Client' extension in VSCode or, if you prefer to talk MQTT, you can publish the json payload to the '/Appliance/UUID _OF_YOUR_DEVICE/subscribe' topic.
Of course these payloads and signs will only work if your device has been reconfigured for local access with a void key (or just put them in pairing mode so the key get reset).
When posting your device response feel free to redact any sensitive information like IPs or UUIDs if you like by replacing them with some dumb recognizable characters

Thank you

**Comment by bytespider** [comment]
I have a copy of the decompiled app that I found helpful in reverse-engineering the messages.

**Reply by krahabb** [reply]
That could be useful, expecially if it's a rather recent one!

I've actually developed most of my code by learning the protocol on https://github.com/albertogeniola/MerossIot.
That is an almost complete protocol implementation with a good representation of all of the devices (didnt check if they were all but they're the most used for sure)
I think an issue would be nevertheless the support of legacy firmwares (like in #15) and I'm actually trying to release an update based on that one but the mobile app could be another 'brick in the wall' (not sure this is the correct meaning tho ;)

If you can share it I'll take a look for sure

**Comment by bytespider** [comment]
[meross_v2.26.2.zip](https://github.com/krahabb/meross_lan/files/6342081/meross_v2.26.2.zip)

The is a lot of code relating to specific MCU features and FW versions. Also quite a bit relating to region subTypes.

**Reply by krahabb** [reply]
Very much appreciated I'll give it a look!
